tests/installation/system_role.pm

```diff
@@ -51,16 +50,20 @@ sub assert_system_role {
     # Still initializing the system at this point, can take some time
     # Asserting screen with preselected role
     # Proper default role assertion will be addressed in poo#37504
-    assert_screen 'system-role-default-system', 180;
-    my $system_role = get_var('SYSTEM_ROLE', 'default');
-    if (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default')) {
-        change_system_role($system_role);
+    # Product might or might not have default selected
+    if (is_opensuse) {
+        assert_screen('before-role-selection', 180);
+        change_system_role(get_var('SYSTEM_ROLE', get_var('DESKTOP')));
```

Comment:
This looks good, but I cannot check if DESKTOP variable is always set (o3 is down atm), so if you add get_var('DESKTOP', 'gnome') it's perfect.

Reply:
I'm not 100% sure that we have DESKTOP always define, as for SLES it's not the case. Right, for openSUSE there is no default, but we still test more on gnome, so in case nothing is given, we can assume it's gnome.

Reply:
@okurz you mean something like this?:
assert_screen [qw(before-role-selection system-role-default-system)], 180;
my $system_role = get_var('SYSTEM_ROLE', get_var('DESKTOP', 'default'));
change_system_role($system_role) unless check_var('SYSTEM_ROLE', 'default');
It might be missing some needle in sle because default instead of gnome is present.

Reply:
I agree with @rwx788 - I would prefer this to be similar in behaviour to SLE

But I really don't think we can afford to wait any longer for these features to start reaching openSUSE, all product def changes have been blocked for weeks now.

I intend to merge this at 1400 unless objections persuade me otherwise